src/NuGet.Core/NuGet.Common/INuGetLock.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Common
+{
+    /// <summary>
+    /// Basic get/release interface for an Async lock.
+    /// </summary>
+    public interface INuGetLock : IDisposable
```

Comment:
1. `IDisposable` is implementation specific and shouldn't be part of the interface declaration.
2. I can't find it's actually used anywhere.